specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+the `ProposalTypeUpdated` event.
+
+#### `execute`
+
+Execute a successful proposal. This MUST only be possible when the quorum is reached, the vote is successful, and the
+deadline is reached.
+
+```solidity
+function execute(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, bytes32 _descriptionHash) external payable returns (uint256)
+```
+
+This function MUST emit the `ProposalExecuted` event and return the proposal ID.
+
+#### `executeWithModule`
+
+Execute a successful proposal that was created with a module. This MUST only be possible when the quorum is reached,
+the vote is successful, and the deadline is reached.
+
+```solidity
+function executeWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash) external payable returns (uint256)
+```
+
+This function MUST emit the `ProposalExecuted` event and return the proposal ID.
+
+#### `propose`
+
+Creates a new proposal with a given proposal type. This function MUST check that the proposal type exists.
+
+```solidity
+function propose(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, string memory _description) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCreated` event and return the ID of the proposal created.
+
+#### `proposeWithModule`
+
+Creates a new proposal with a given proposal type and module. This function MUST check that the proposal type exists,
+and that the module is supported by the `Governor` contract.
+
+```solidity
+function proposeWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash, uint8 _proposalType) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCreated` event and return the ID of the proposal created.
+
+#### `relay`
+
+Relays a transaction or function call to an arbitrary target. This function MUST only be callable as part of a
+governance proposal.
+
+```solidity
+function relay(address _target, uint256 _value, bytes calldata _data) external payable
+```
+
+#### `setModuleApproval`
+
+Approve or reject a module. This function MUST only be callable by the manager.
+
+```solidity
+function setModuleApproval(address _module, bool _approved) external
+```
+
+#### `setProposalDeadline`
+
+Update the deadline timestamp of an active proposal. This function MUST only be callable by the manager.
+
+```solidity
+function setProposalDeadline(uint256 _proposalId, uint64 _deadline) external
+```
+
+This function MUST emit the `ProposalDeadlineUpdated` event.
+
+#### `setProposalThreshold`
+
+Updates the proposal threshold. This function MUST only be callable by the manager.
+
+```solidity
+function setProposalThreshold(uint256 _newProposalThreshold) external
+```
+
+This function MUST emit the `ProposalThresholdSet` event.
+
+#### `setVotingDelay`
+
+Updates the voting delay. This function MUST only be callable by the manager.
```

Comment:
What is the voting delay?

Reply:
specified in specified in https://github.com/ethereum-optimism/specs/pull/315/commits/9a0fb5809772b4a6840fb165b1d4579658b6c7cb